File: core/src/main/java/me/devsnox/planetsystem/core/commands/modules/SethomeCommand.java
package me.devsnox.planetsystem.core.commands.modules;

import me.devsnox.planetsystem.api.holder.Holder;
import me.devsnox.planetsystem.api.location.PlanetLocation;
import me.devsnox.planetsystem.api.player.PlanetPlayer;
import me.devsnox.planetsystem.core.commands.PlanetCommandModule;

public class SethomeCommand implements PlanetCommandModule {

    @Override
    public void execute(final PlanetPlayer player, final String[] args) {
        if (player.getPlayer().getLocation().getWorld().equals(Holder.Impl.holder.getWorld())) {
            player.getPlanet().setSpawnLocation(PlanetLocation.create(player.getPlayer().getLocation(), player.getPlanet()));
            player.getLogger().info("commands.sethome.success");
        } else {
            player.getLogger().warn("commands.sethome.not_in_planet_world");
        }
    }
}

File: core/src/main/kotlin/me/devsnox/planetsystem/core/commands/modules/SethomeCommand.kt
package me.devsnox.planetsystem.core.commands.modules

import me.devsnox.planetsystem.api.holder.isNotInHolderWorld
import me.devsnox.planetsystem.api.location.PlanetLocation
import me.devsnox.planetsystem.api.player.PlanetPlayer
import me.devsnox.planetsystem.core.commands.PlanetCommandModule
import me.devsnox.planetsystem.core.log.MessageKeys.*

class SethomeCommand : PlanetCommandModule {

	override fun execute(player: PlanetPlayer, args: Array<String>) = if (player.player.isNotInHolderWorld())
		player.logger.warn(COMMANDS_SET_HOME_NOT_IN_PLANET_WORLD)
	else {
		player.planet.spawnLocation = PlanetLocation(player.planet, player.player.location)
		player.logger.info(COMMANDS_SET_HOME_SUCCESSES)
	}

}
